apps/emqx_auth_jwt/src/emqx_auth_jwt.erl

```diff
@@ -75,14 +73,40 @@ check_acl(ClientInfo = #{jwt_claims := Claims},
     end.
 
 is_expired(Exp) when is_binary(Exp)  ->
-    ExpInt  = binary_to_integer(Exp),
-    is_expired(ExpInt);
-is_expired(Exp) ->
+    case binary_to_number(Exp) of
+        {ok, Val} ->
+            is_expired(Val);
+        _ ->
+            ?DEBUG("acl_deny_due_to_invalid_jwt_exp:~p", [Exp]),
+            true
+    end;
+is_expired(Exp) when is_integer(Exp) ->
     Now = erlang:system_time(second),
-    Now > Exp.
+    Now > Exp;
+is_expired(Exp) ->
+    ?DEBUG("acl_deny_due_to_invalid_jwt_exp:~p", [Exp]),
+    true.
 
 description() -> "Authentication with JWT".
 
+binary_to_number(Bin) ->
+    Checker = fun([], _) ->
+                      false;
+                 ([H | T], Self) ->
+                      try
+                           {ok, H(Bin)}
+                      catch _:_ ->
+                              Self(T, Self)
+                      end
+              end,
+
+    Checker([fun erlang:binary_to_integer/1,
+            fun(In) ->
+                    Val = erlang:binary_to_float(In),
+                    erlang:round(Val)
+            end],
+           Checker).
+
```

Comment:
nit: we can give a name to the func. it is better style, I think.
```erlang
binary_to_number(Bin) when is_binary(Bin) ->
  Apply = fun _Apply([]) -> false;
              _Apply([Fun | More]) -> try {ok, Fun(Bin)} catch _:_ -> _Apply(More)
                end,
  Apply([fun ...]).
```

Reply:
This has been simplified